llvm/lib/Target/EraVM/EraVMInstrFormats.td

```diff
@@ -697,7 +697,7 @@ class IFarCall<EraVMOpcode opcode,
       !cond(!eq({is_static, is_shard}, {0, 0}): "",
             !eq({is_static, is_shard}, {1, 0}): ".static",
```

Comment:
Does it mean that these full notations are also accepted?

Reply:
Strictly speaking, now the syntax is not fully consistent: everything uses the old names except for the combined `.st.sh` modifier (it did not exist before #539). Thus, exactly the following forms are accepted (`far_call` can be replaced with `far_call.delegate` or `far_call.mimic`):
```
  far_call        r1, r2, @foo
  far_call.static r1, r2, @foo
  far_call.shard  r1, r2, @foo
  far_call.st.sh  r1, r2, @foo
  far_call.sh.st  r1, r2, @foo ; supported via MnemonicAlias
; the same with predicate, such as far_call.shard.lt r1, r2, @foo
```
Note that #539 replaced conditional concatenation of `.static` and then `.shard` with 2x2 cartesian product because `.{st,sh}` was expected in the new syntax that cannot be represented that way.

I can simplify this back to something like
```
let AsmString = !strconcat(opcode.Name,
                           !if(is_shard, ".shard", ""),
                           !if(is_static, ".static", ""),
                           "${cc}", "\t", asmstr);
```
In this case the fourth variant (`.st.sh`) would be `.static.shard` and will be renamed in #576. Or we can postpone this simplification until #576 is merged.

Reply:
Understood.
Sounds like the legacy is piling up, so we need to migrate to the new syntax completely.

Reply:
Let's address it in another PR.
Here LGTM!

Reply:
Do you mean replacing `.static` and `.shard` with `.st` and `.sh`? This could break CI if these modifiers are actually emitted for system contracts, for example. My idea was to either rollback `.st.sh` to `.static.shard` temporarily (and simplify `AsmString` computation now) or do not clutter this PR and #576 by renaming modifiers back-and-forth and postpone simplification of `AsmString`.